### DISCUSSION TITLE
Why doesn't Preact allow adding an extended signal as children?

### DISCUSSION THREAD

**Discussion by denisavitski** [preactjs/signals#465] [Q&A]
Like so:

```ts
class MySignal<T> extends Signal<T> {
  constructor(value: T) {
    super(value)
  }
}

const mySignal = new MySignal(0)

export const App = () => {
  return <div>{mySignal}</div>
}
```

Uncaught Error: Objects are not valid as a child.

**Comment by rschristian** [comment]
What version of Preact are you using? Should work fine: https://stackblitz.com/edit/vitejs-vite-yuwwvx?file=package.json

**Reply by denisavitski** [reply]
I found out that in version 10.19.3 this does not work, but in 10.19.2 it works well

**Reply by rschristian** [reply]
Ah, indeed. Looks like this was an intentional fix recently: https://github.com/preactjs/preact/pull/4219#issuecomment-1827610946

Might want to ask on the Preact repo, I'm not aware of the context there myself.
